packages/material-ui/src/Unstable_TrapFocus/Unstable_TrapFocus.js

```diff
@@ -85,6 +87,19 @@ function Unstable_TrapFocus(props) {
       }
 
       if (rootRef.current && !rootRef.current.contains(doc.activeElement)) {
+        // if the focus event is not coming from inside the children's react tree, reset the refs
+        if (
+          (nativeEvent && reactFocusEventTarget.current !== nativeEvent.target) ||
+          doc.activeElement !== reactFocusEventTarget.current
+        ) {
+          insideReactTreeRef.current = false;
+          reactFocusEventTarget.current = null;
+        }
+
+        const insideReactTree = insideReactTreeRef.current;
```

Comment:
I assume this was done for readability but since isn't now the same as the ref name we don't really need it anymore.

Reply:
What are your thoughts on changing it to `insideReactTree` as part of this change too?

Reply:
Sorry I don’t really understand the suggestion. Are you saying we should get rid of the extra const? If that is your comment I agree, it’s redundant now with the renaming done :)

Reply:
Exactly that :)